scripts/check-consensus-builds-with-chap.sh

```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+
+set -o pipefail
+
+# remove network packages from the list of local packages
+
+sed 's/\.\/.*network.*//g' cabal.project \
```

Comment:
This `sed` line seems practical, but it also seems like something that will often cause confusion.

For example, I think a better solution would be:

- enumerate the packages we want to exclude
- enumerate the packages we want to include
- fail if the packages we find in the actual file are not equal the union of the above
- generate the new file by removing the ones we want to exclude

Reply:
Reasonable, let me rework it

Reply:
Another alternative: move the `packages:` part to two standalone files (one for network, one for consensus) and [import them](https://cabal.readthedocs.io/en/latest/cabal-project.html#conditionals-and-imports), then you can replace the network file with an empty file here temporarily.